markdown/spring-cloud-stream-beyond/spring-cloud-stream-beyond.md

```diff
@@ -0,0 +1,999 @@
+``` yaml
+spring:
+  cloud:
+    function:
+      definition: myConsumer
+    stream:
+      bindings:
+        myConsumer-in-0:
+          destination: 'spring/*/stream/>'
+          group: nonexclusive
+          consumer:
+            concurrency: 5
+      solace:
+        bindings:
+          myConsumer-in-0:
+            consumer:
+              queueAdditionalSubscriptions:
+                - 'a/b/>'
+                - 'pub/*/plus'
+```
+
+After restarting our app we can see that the subscriptions on our queue have been updated to include our two additions:
+![Wildcard Queue 2](img/wildcardQueue2.webp)
+
+💥 We can now use Solace wildcards to filter for the exact events that we're interested in! 
+
+## Dynamic Publishing
+Duration: 0:15:00
+
+Now that we know how to subscribe using wildcards we're halfway to taking advantage of Solace's dynamic topics. The second part of that is of course to be able to publish to whatever topic we need to! We want to publish to a unique topic for each event, no worries I've got you covered! 
+
+The Spring Cloud Stream framework allows for 2 different ways to publish to dynamic topics using the imperative style.  
+1. Using `StreamBridge`
+1. Using the `BinderHeaders.TARGET_DESTINATION` header
+
+Negative
+: If you're using the reactive programming style you can use the `EmitterProcessor` for dynamic publishing but we won't be covering that in this codelab. 
+
+### StreamBridge
+The first option for dynamic publishing is using `StreamBridge`. Note that this option is processed at the framework level and will work with any Cloud Stream binder that you choose to use. `StreamBridge` will cache a channel within Spring for each destination that you publish to. This options is solid if you're going to be publishing to a small number of destinations as the channel will remain in cache and you can lookup the channel for monitoring/metrics if you desire. Note that the number of channels cached is configurable via `spring.cloud.stream.dynamic-destination-cache-size`. 
+
+Let's update our `myConsumer` function to publish to dynamic topics. 
+``` java
+@Bean
+public Consumer<Message<String>> myConsumer(StreamBridge sb) {
+    return v -> {
+        logger.info("Received myConsumer: " + v.getPayload());
+        logger.info("CorrelationID: " + v.getHeaders().get("solace_correlationId"));
+        
+        // Use whatever business logic you'd like to figure out the topic!
+        String cid = (String) v.getHeaders().get("solace_correlationId");
+        String myTopic = "solace/cid/".concat(cid);
+        logger.info("Publishing to: " + myTopic);
+        sb.send(myTopic , v.getPayload());
+    };
+}
+```
+
+Positive
+: Note that StreamBridge does not send the output of the function as message, but rather can be used to send a message whenever needed during processing. This can come in handle to send alerts during processing.
```

Comment:
Typo: This can come in `handle` to send alerts during processing

Reply:
thanks